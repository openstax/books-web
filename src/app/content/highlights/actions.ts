import { createStandardAction } from 'typesafe-actions';

export const enableHighlighting = createStandardAction('FeatureFlag/enableHighlighting')<void>();
