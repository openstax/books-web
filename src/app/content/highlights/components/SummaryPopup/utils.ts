import { Highlight } from '@openstax/highlighter/dist/api';
import queryString from 'querystring';
import { Book } from '../../../types';
import { findArchiveTreeNodeById } from '../../../utils/archiveTreeUtils';
import { getBookPageUrlAndParams } from '../../../utils/urlUtils';

export const createHighlightLink = ({ sourceId, id, anchor: elementId }: Highlight, book?: Book) => {
  const page = book ? findArchiveTreeNodeById(book.tree, sourceId) : undefined;
  const data = { target: JSON.stringify({ id, type: 'highlight' }) };
  const target = queryString.stringify(data) + `#${elementId}`;
  return `${page && book ? getBookPageUrlAndParams(book, page).url : ''}?${target}`;
};
