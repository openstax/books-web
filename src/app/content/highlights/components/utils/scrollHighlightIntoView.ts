import { Highlight } from '@openstax/highlighter';
import { HTMLElement } from '@openstax/types/lib.dom';
import { scrollIntoView } from '../../../../domUtils';

export default (highlight: Highlight, element?: HTMLElement | null) => {
  const firstElement = highlight.elements[0] as HTMLElement;
  const lastElement = highlight.elements[highlight.elements.length - 1] as HTMLElement;
  const otherElements = [lastElement];
  if (element) {
    otherElements.push(element);
  }
  scrollIntoView(firstElement, otherElements);
};
