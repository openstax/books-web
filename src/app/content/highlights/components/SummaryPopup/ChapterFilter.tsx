import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled, { css } from 'styled-components/macro';
import AllOrNone from '../../../../components/AllOrNone';
import Checkbox from '../../../../components/Checkbox';
import { textStyle } from '../../../../components/Typography/base';
import { match, not } from '../../../../fpUtils';
import theme from '../../../../theme';
import { filters } from '../../../styles/PopupConstants';
import { setSummaryFilters } from '../../actions';
import { highlightLocationFilters, highlightLocationFiltersWithContent, summaryLocationFilters } from '../../selectors';
import ColorIndicator from '../ColorIndicator';

interface Props {
  className?: string;
}

// tslint:disable-next-line:variable-name
const Row = styled.div`
  display: flex;
  flex-direction: row;
  ${theme.breakpoints.mobile(css`
    flex-direction: column;
    overflow: hidden;
  `)}
`;

// tslint:disable-next-line:variable-name
const Column = styled.div`
  display: flex;
  flex-direction: column;
`;

// tslint:disable-next-line:variable-name
const ChapterTitle = styled.span`
  display: flex;
  flex-direction: row;
  white-space: nowrap;
  margin-left: 0.8rem;

  > * {
    overflow: hidden;
  }

  .os-text {
    flex: 1;
    text-overflow: ellipsis;
  }

  .os-divider {
    margin: 0 0.4rem;
  }
`;

const chunk = <T extends any>(sections: T[]) => {
  const cutoff = Math.max(20, Math.ceil(sections.length / 2));
  return [sections.slice(0, cutoff), sections.slice(cutoff)].filter((arr) => arr.length > 0);
};

// tslint:disable-next-line:variable-name
const ChapterFilter = ({className}: Props) => {
  const locationFilters = useSelector(highlightLocationFilters);
  const locationFiltersWithContent = useSelector(highlightLocationFiltersWithContent);
  const selectedLocationFilters = useSelector(summaryLocationFilters);
  const dispatch = useDispatch();

  const setSelectedChapters = (ids: string[]) => {
    dispatch(setSummaryFilters({locationIds: ids}));
  };

  const handleChange = (id: string) => {
    if (selectedLocationFilters.has(id)) {
      setSelectedChapters([...selectedLocationFilters].filter(not(match(id))));
    } else {
      setSelectedChapters([...selectedLocationFilters, id]);
    }
  };

  return <div className={className} tabIndex={-1}>
    <AllOrNone
      onNone={() => setSelectedChapters([])}
      onAll={() => setSelectedChapters(Array.from(locationFiltersWithContent))}
    />
    <Row>
      {chunk(Array.from(locationFilters.values())).map((sectionChunk, index) => <Column key={index}>
        {sectionChunk.map((location) => <Checkbox
          key={location.id}
          checked={selectedLocationFilters.has(location.id)}
          disabled={!locationFiltersWithContent.has(location.id)}
          onChange={() => handleChange(location.id)}
        >
          <ChapterTitle dangerouslySetInnerHTML={{__html: location.title}} />
        </Checkbox>)}
      </Column>)}
    </Row>
  </div>;
};

export default styled(ChapterFilter)`
  ${textStyle}
  background: ${theme.color.white};
  font-size: 1.4rem;
  padding: ${filters.dropdownContent.padding.topBottom}rem ${filters.dropdownContent.padding.sides}rem;
  outline: none;
  overflow: auto;
  z-index: 1;
  ${theme.breakpoints.mobile(css`
    &&& {
      max-width: calc(100vw - ${filters.valueToSubstractFromVW}rem);
    }
  `)}

  ${AllOrNone} {
    margin: 0.8rem 0 0.8rem 0.8rem;
  }

  ${Checkbox} {
    padding: 0.8rem;
  }

  ${ColorIndicator} {
    margin: 0 1.6rem 0 1.6rem;
  }
`;
