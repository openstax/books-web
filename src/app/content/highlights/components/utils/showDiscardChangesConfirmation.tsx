import { MiddlewareAPI } from '../../../../types';
import { showConfirmationModal } from '../../../actions';
import { setForceScrollToHiglight } from '../../actions';

export default (
  dispatch: MiddlewareAPI['dispatch']): Promise<boolean> => new Promise((resolve) => {
    dispatch(showConfirmationModal({
      options: {
        bodyi18nKey: 'i18n:discard:body',
        callback: (confirmed: boolean) => {
          resolve(confirmed);
          if (!confirmed) {
            dispatch(setForceScrollToHiglight(true));
          }
        },
        cancelButtoni18nKey: 'i18n:discard:button:cancel',
        headingi18nKey: 'i18n:discard:heading',
        okButtoni18nKey: 'i18n:discard:button:discard',
      },
    }));
  });
