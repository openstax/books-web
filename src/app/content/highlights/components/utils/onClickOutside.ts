import { EventListenerOptions, HTMLElement, MouseEvent } from '@openstax/types/lib.dom';
import React from 'react';
import { assertDocument, assertWindow } from '../../../../utils';

const onClickOutside = (
  element: React.RefObject<HTMLElement>,
  isFocused: boolean,
  cb: (ev: MouseEvent) => void,
  eventOptions?: EventListenerOptions
) => () => {
  if (typeof document === 'undefined') {
    return () => null;
  }

  const ifOutside = (e: MouseEvent) => {
    if (!(e.target instanceof assertWindow().Element)) {
      return;
    }
    if (!element.current || element.current.contains(e.target)) {
      return;
    }
    cb(e);
  };

  if (isFocused) {
    document.addEventListener('click', ifOutside, eventOptions);
  }

  return () => assertDocument().removeEventListener('click', ifOutside, eventOptions);
};

export const useOnClickOutside = (
  element: React.RefObject<HTMLElement>,
  isEnabled: boolean,
  cb: (e: MouseEvent) => void,
  eventOptions?: EventListenerOptions
) => {
  React.useEffect(onClickOutside(element, isEnabled, cb, eventOptions), [isEnabled]);
};

export default onClickOutside;
