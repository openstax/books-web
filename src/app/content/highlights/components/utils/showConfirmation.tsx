import React from 'react';
import ReactDOM from 'react-dom';
import uuid from 'uuid/v4';
import TestContainer from '../../../../../test/TestContainer';
import { assertDocument, assertNotNull } from '../../../../utils';
import ConfirmationModal from '../ConfirmationModal';

export default async() => {
  const document = assertDocument();
  const domNode = document.createElement('div');

  domNode.id = `dialog-${uuid()}`;
  const root = assertNotNull(document.getElementById('root'), 'root element not found');
  root.insertAdjacentElement('afterend', domNode);

  const userChoice: boolean = await new Promise((resolve) => {
    const confirm = () => resolve(true);
    const deny = () => resolve(false);
    ReactDOM.render(
      <TestContainer>
        <ConfirmationModal deny={deny} confirm={confirm} />
      </TestContainer>,
      domNode
    );
  });

  ReactDOM.unmountComponentAtNode(domNode);
  assertNotNull(domNode.parentElement, 'parent element not found').removeChild(domNode);

  return userChoice;
};
