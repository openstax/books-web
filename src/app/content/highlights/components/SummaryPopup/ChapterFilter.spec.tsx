import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestServices from '../../../../../test/createTestServices';
import createTestStore from '../../../../../test/createTestStore';
import { book as archiveBook, page, shortPage } from '../../../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../../../test/mocks/osWebLoader';
import AllOrNone from '../../../../components/AllOrNone';
import Checkbox from '../../../../components/Checkbox';
import MessageProvider from '../../../../MessageProvider';
import { MiddlewareAPI, Store } from '../../../../types';
import { receiveBook, receivePage } from '../../../actions';
import { formatBookData } from '../../../utils';
import { setHighlightsTotalCountsPerLocation } from '../../actions';
import { addCurrentPageToSummaryFilters } from '../../utils';
import ChapterFilter from './ChapterFilter';

describe('ChapterFilter', () => {
  const book = formatBookData(archiveBook, mockCmsBook);
  let store: Store;
  let helpers: ReturnType<typeof createTestServices> & MiddlewareAPI;

  beforeEach(() => {
    store = createTestStore();

    helpers = {
      ...createTestServices(),
      dispatch: store.dispatch,
      getState: store.getState,
    };

    store.dispatch(receivePage({...page, references: []}));
  });

  it('matches snapshot', () => {
    store.dispatch(receiveBook(book));
    store.dispatch(setHighlightsTotalCountsPerLocation({
      'testbook1-testpage1-uuid': 1,
    }));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without a book', () => {
    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const checkedBoxes = component.root.findAllByProps({checked: true});
    expect(checkedBoxes.length).toBe(0);
  });

  it('initially has the selected page checked', () => {
    store.dispatch(receiveBook(book));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const [box1] = component.root.findAllByType(Checkbox);

    expect(box1.props.checked).toBe(true);
  });

  it('initially has the selected chapter checked', () => {
    store.dispatch(receiveBook(book));
    store.dispatch(receivePage({...shortPage, references: []}));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const [, , , , box5] = component.root.findAllByType(Checkbox);

    expect(box5.props.checked).toBe(true);
  });

  it('unchecks chapters', () => {
    store.dispatch(receiveBook(book));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const [box1, box2] = component.root.findAllByType(Checkbox);

    expect(box1.props.checked).toBe(true);
    expect(box2.props.checked).toBe(false);

    renderer.act(() => {
      box1.props.onChange();
    });

    expect(box1.props.checked).toBe(false);
    expect(box2.props.checked).toBe(false);
  });

  it('checks chapters', () => {
    store.dispatch(receiveBook(book));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const [box1, box2] = component.root.findAllByType(Checkbox);

    expect(box1.props.checked).toBe(true);
    expect(box2.props.checked).toBe(false);

    renderer.act(() => {
      box2.props.onChange();
    });

    expect(box1.props.checked).toBe(true);
    expect(box2.props.checked).toBe(true);
  });

  it('selects none', () => {
    store.dispatch(receiveBook(book));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const [box1, box2] = component.root.findAllByType(Checkbox);
    const allOrNone = component.root.findByType(AllOrNone);

    expect(box1.props.checked).toBe(true);
    expect(box2.props.checked).toBe(false);

    renderer.act(() => {
      allOrNone.props.onNone();
    });

    expect(box1.props.checked).toBe(false);
    expect(box2.props.checked).toBe(false);
  });

  it('selects all select only chapters with highlights', () => {
    store.dispatch(receiveBook(book));
    store.dispatch(setHighlightsTotalCountsPerLocation({
      'testbook1-testpage1-uuid': 1,
    }));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const [box1, box2] = component.root.findAllByType(Checkbox);
    const allOrNone = component.root.findByType(AllOrNone);

    renderer.act(() => {
      allOrNone.props.onNone();
    });

    expect(box1.props.checked).toBe(false);
    expect(box2.props.checked).toBe(false);

    renderer.act(() => {
      allOrNone.props.onAll();
    });

    expect(box1.props.checked).toBe(true);
    expect(box2.props.checked).toBe(false);
  });

  it('chapters without highlights are disabled', () => {
    store.dispatch(receiveBook(book));
    store.dispatch(setHighlightsTotalCountsPerLocation({
      'testbook1-testpage1-uuid': 1,
    }));
    addCurrentPageToSummaryFilters(helpers);

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <ChapterFilter />
      </MessageProvider>
    </Provider>);

    const [box1, box2] = component.root.findAllByType(Checkbox);

    expect(box1.props.disabled).toBe(false);
    expect(box2.props.disabled).toBe(true);
  });
});
