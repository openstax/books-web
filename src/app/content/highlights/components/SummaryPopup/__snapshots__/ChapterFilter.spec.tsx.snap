// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterFilter matches snapshot 1`] = `
.c13 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c4 {
  cursor: pointer;
  border: none;
  padding: 0;
  background: none;
  outline: none;
  font-family: Neue Helvetica W01;
  color: #424242;
  color: #027EB5;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover,
.c4:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #0064A0;
}

.c2 {
  height: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2,
.c2 .c3 {
  font-size: 1.4rem;
}

.c2 span {
  padding: 0 1rem;
}

.c12 {
  color: #fff;
  height: 1rem;
  width: 1rem;
  display: none;
}

.c10 {
  height: 1.6rem;
  width: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

input:not(:checked) + .c10 {
  border: 1px solid #424242;
}

input:checked + .c10 {
  background-color: #df571e;
}

input:checked + .c10 .c11 {
  display: block;
}

.c8 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.c8:focus-within {
  border-radius: 0.4rem;
  background-color: #f1f1f1;
}

.c8:focus-within .c9 {
  border: 1px solid #df571e;
}

.c8.focus-within {
  border-radius: 0.4rem;
  background-color: #f1f1f1;
}

.c8.focus-within .c9 {
  border: 1px solid #df571e;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  white-space: nowrap;
  margin-left: 0.8rem;
}

.c14 > * {
  overflow: hidden;
}

.c14 .os-text {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-overflow: ellipsis;
}

.c14 .os-divider {
  margin: 0 0.4rem;
}

.c0 {
  font-family: Neue Helvetica W01;
  color: #424242;
  background: #fff;
  font-size: 1.4rem;
  padding: 0.8rem 1.6rem;
  outline: none;
  max-height: 72rem;
  overflow: auto;
}

.c0 .c1 {
  margin: 0.8rem 0 0.8rem 0.8rem;
}

.c0 .c7 {
  padding: 0.8rem;
}

.c0 .c15 {
  margin: 0 1.6rem 0 1.6rem;
}

@media screen and (max-width:75em) {
  .c5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    overflow: hidden;
  }
}

@media screen and (max-width:75em) {
  .c0.c0.c0 {
    left: -1.6rem;
    max-width: calc(100vw - 0.8rem * 2);
  }
}

<div
  className="c0"
  tabIndex={-1}
>
  <div
    className="c1 c2"
  >
    <button
      className="c3 c4"
      onClick={[Function]}
    >
      All
    </button>
    <span>
      |
    </span>
    <button
      className="c3 c4"
      onClick={[Function]}
    >
      None
    </button>
  </div>
  <div
    className="c5"
  >
    <div
      className="c6"
    >
      <label
        className="c7 c8"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-text\\">Test Page 1</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 1</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 1.1</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">2</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 2</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">3</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 3</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">4</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 4</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">10</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 5</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">12</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 6</span>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">A</span><span class=\\"os-divider\\"> | </span><span class=\\"os-text\\">Atomic Masses</span></a>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-number\\">D</span><span class=\\"os-divider\\"> | </span><span class=\\"os-text\\">Glossary of Key Symbols and Notation</span></a>",
            }
          }
        />
      </label>
      <label
        className="c7 c8"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c9 c10"
        >
          <svg
            aria-hidden="true"
            className="c11 c12 c13"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 512 512"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          className="c14"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-text\\">Test Page for Generic Styles</span>",
            }
          }
        />
      </label>
    </div>
  </div>
</div>
`;
